lib/processors/lib/utils.py

```diff
@@ -36,20 +37,26 @@ def make_handler(name, handler, checkauth):
     class ProcAPIHandler(BaseHandler):
         @gen.coroutine
         def process_request(self):
-            if checkauth:
-                userid = self.get_argument('userid')
-                exibperm = yield ExhibitPermissions.get_global()
-                permission = yield exibperm.has_permission(userid, name)
-                if permission is not True:
-                    return self.error(403, "NO_ACCESS: {}".format(userid))
-            else:
-                userid = self.get_argument('userid', None)
+            userid = self.get_argument('userid', None)
+            rfid = self.get_argument('rfid', None)
+            if checkauth and not (userid is None) ^ (rfid is None):
```

Comment:
`^` is xor (exclusive or).... basically it checks if userid is none or rfid is none but not both.